packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -320,7 +333,13 @@ export async function runHttpQuery(
         fieldResolver: optionsObject.fieldResolver,
         debug: optionsObject.debug,
         tracing: optionsObject.tracing,
-        cacheControl: optionsObject.cacheControl,
+        //we always want cacheControl to either set the CDN headers or for the
+        //engine proxy
+        cacheControl:
+          optionsObject.cacheControl !== null &&
```

Comment:
hmm, why not just `optionsObject.cacheControl === undefined ? true : optionsObject.cacheControl`? Looks like this will override explicitly `cacheControl: false` which seems wrong.